lib/puppet/parser/functions/unique.rb

```diff
@@ -1,51 +0,0 @@
-# frozen_string_literal: true
-
-#
-# unique.rb
-#
-module Puppet::Parser::Functions
-  newfunction(:unique, type: :rvalue, doc: <<-DOC
-    @summary
-      This function will remove duplicates from strings and arrays.
-
-    @return
-      String or array with duplicates removed
-
-    @example **Usage**
-
-      unique("aabbcc")
-      Will return: abc
-
-      You can also use this with arrays:
-
-      unique(["a","a","b","b","c","c"])
-      This returns: ["a","b","c"]
-
-    DOC
-  ) do |arguments|
-    if Puppet::Util::Package.versioncmp(Puppet.version, '5.0.0') >= 0
-      function_deprecation([:unique, 'This method is deprecated, please use the core puppet unique function. There is further documentation for the function in the release notes of Puppet 5.0.'])
```

Comment:
This is intriguing.  Was this ever reachable??  ie If you were on Puppet 5 wouldn't the built in function automatically get used instead and was there anyway round this?